index.js

```diff
@@ -171,22 +174,35 @@ async function initLastProcessedLedger() {
   }
 }
 
-const fetchEvents = () => {
-  return work()
-    .then(() => {
-      logger.info(`Progressed to position ${JSON.stringify(lastProcessedPosition)}`)
+async function fetchEvents() {
+  try {
+    await work()
+  }
+  catch(ex) {
+    logger.error(ex)
+    // This can throw if no more API endpoints left
+    await createNewSetConnections();
+  }
 
-      // Look for new events every 1 sec
-      setTimeout(fetchEvents, 1000)
-    })
+  logger.info(`Progressed to position ${JSON.stringify(lastProcessedPosition)}`)
+
+  // Look for new events every 1 sec
+  setTimeout(fetchEvents, 1000)
 }
 
-const init = async () => {
-  metrics.startCollection()
+async function createNewSetConnections() {
+  var nodeURLs = XRP_NODE_URLS.split(",");
```

Comment:
What do you think about constructing global nodeURLs at the start and just pop the next url in this method, so we can exclude var `apiURLToUse`? IMHO it makes code a bit simpler.

Reply:
Yes, much better indeed! Fixed.